src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9719,13 +9756,16 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                                 argumentSyntax, singleCandidate);
                         }
                     }
-                    else
+                    // For C# 12 and earlier statically bind invocations in presence of dynamic arguments only for expanded non-array params cases.
```

Comment:
I'm confused, did we support non-array params cases in C# 12 or earlier?

Reply:
> did we support non-array params cases in C# 12 or earlier? Also applies to other places that check language version

We didn't. In general we prefer to not change semantic analysis based on language version. And I prefer to keep it this way specifically for params collections. Using them in C# 12 is an error, but we still force C# 13 binding.